<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MaterialEditor</class>
 <widget class="QWidget" name="MaterialEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>671</width>
    <height>591</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>ambient color</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QDoubleSpinBox" name="a0_0">
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
         <property name="value">
          <double>0.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="a0_1">
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QDoubleSpinBox" name="a0_3">
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QDoubleSpinBox" name="a0_2">
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>diffuse color</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QDoubleSpinBox" name="d0_0">
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
         <property name="value">
          <double>0.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="d0_1">
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QDoubleSpinBox" name="d0_3">
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QDoubleSpinBox" name="d0_2">
         <property name="decimals">
          <number>3</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.001000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>specular color</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="s0_0">
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>0.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
       <property name="value">
        <double>0.000000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="s0_1">
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>0.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="s0_2">
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>0.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="s0_3">
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>0.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>shininess</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="shininess">
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>ambient coefficient</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="acoeff">
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>diffuse coefficient</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_7">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="dcoeff">
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_8">
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>specular coefficient</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_8">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="scoeff">
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="minimum">
        <double>-99.989999999999995</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
  <zorder>label</zorder>
  <zorder>label_2</zorder>
  <zorder>label_3</zorder>
  <zorder>label_4</zorder>
  <zorder>gridLayoutWidget</zorder>
  <zorder>gridLayoutWidget_2</zorder>
  <zorder>shininess</zorder>
  <zorder>gridLayoutWidget_3</zorder>
  <zorder>horizontalSpacer</zorder>
  <zorder>horizontalSpacer_2</zorder>
  <zorder>horizontalSpacer_3</zorder>
  <zorder>horizontalSpacer_4</zorder>
  <zorder>layoutWidget</zorder>
  <zorder>layoutWidget_2</zorder>
  <zorder>label_5</zorder>
  <zorder>label_7</zorder>
  <zorder>layoutWidget_3</zorder>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>a0_0</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>354</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>702</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>a0_1</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>448</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>744</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>a0_2</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>517</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>723</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>a0_3</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>588</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>683</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>d0_0</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>366</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>704</x>
     <y>296</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>d0_1</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>451</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>666</x>
     <y>326</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>d0_2</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>533</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>653</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>d0_3</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>600</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>679</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>s0_0</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>675</x>
     <y>427</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>s0_1</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>437</x>
     <y>206</y>
    </hint>
    <hint type="destinationlabel">
     <x>670</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>s0_2</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>670</x>
     <y>421</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>s0_3</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>636</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>660</x>
     <y>453</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shininess</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>616</x>
     <y>294</y>
    </hint>
    <hint type="destinationlabel">
     <x>668</x>
     <y>562</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>acoeff</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>590</x>
     <y>380</y>
    </hint>
    <hint type="destinationlabel">
     <x>700</x>
     <y>402</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dcoeff</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>596</x>
     <y>459</y>
    </hint>
    <hint type="destinationlabel">
     <x>697</x>
     <y>500</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scoeff</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MaterialEditor</receiver>
   <slot>updateSceneObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>588</x>
     <y>543</y>
    </hint>
    <hint type="destinationlabel">
     <x>689</x>
     <y>588</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>updateObject(double)</signal>
  <slot>updateSceneObject()</slot>
 </slots>
</ui>
